### PR TITLE
[50961] Ensure computation of the derived values is triggered at the right time

### PR DESCRIPTION
**Pull request by cbliard**
https://community.openproject.org/wp/50961

The following triggers where missing to trigger a recalculation of the derived values:
- deleting a child
- changing status to closed/open

### CODE REVIEW

File: app/services/work_packages/update_ancestors_service.rb [hunk 1]
```diff
@@ -212,7 +212,14 @@ def all_remaining_hours(work_packages)
...
   def modified_attributes_justify_derivation?(attributes)
-    attributes.intersect?(%i[estimated_hours done_ratio parent parent_id ignore_non_working_days remaining_hours])
+    attributes.intersect?(%i[
+                            done_ratio
+                            estimated_hours
+                            ignore_non_working_days
+                            parent parent_id
+                            remaining_hours
+                            status status_id
+                          ])
```
Comment (aaron-contreras) [inline app/services/work_packages/update_ancestors_service.rb:222]: Alphabetized, nice.

File: spec/features/work_packages/bulk/copy_work_package_spec.rb [hunk 2]
```diff
@@ -312,7 +312,8 @@
 
       it 'does not allow to copy' do
         context_menu.open_for work_package
-        context_menu.expect_no_options ['Bulk copy']
+        $stop = true
```
Comment (aaron-contreras) [inline spec/features/work_packages/bulk/copy_work_package_spec.rb]: Were you using this to debug?

File: spec/support/rspec_retry.rb [hunk 3]
```diff
@@ -29,7 +29,17 @@
...
+  # `rand_factor`, and `multiplier` parameters and thus ruins the benefit of
+  # calling `retry_block` with `args: { tries: _ }` argument.
+  #
+  # Prefer setting `base_interval`, `max_interval`, `rand_factor`, and
+  # `multiplier` instead to keep the benefit of `args: { tries: _ }` argument.
+  #
+  # This will generate the following intervals: [0.5, 0.75, 1.125, ~1.7, ~2.5, ~3.8, ...]
+  c.base_interval = 0.5
+  c.multiplier = 1.5
+  c.rand_factor = 0.0
```
Comment (aaron-contreras) [inline spec/support/rspec_retry.rb:42]: 🤌

File: spec/support/edit_fields/edit_field.rb [hunk 4]
```diff
@@ -10,17 +10,35 @@ class EditField
...
+  # work package creation page (split or full).
+  #
+  # @param context [Object] The context in which the EditField is being used.
+  # @param property_name [Symbol] The name of the property associated with the
+  #   EditField. Generally camel case.
+  # @param selector [String] (optional) The CSS selector used to locate the
+  #   EditField element. if unspecified, the `property_name` is used.
+  # @param create_form [Boolean] (optional) Indicates whether the EditField is
+  #   used in a create form. It changes the way the field is clicked or not to
+  #   activate it on edition. It is `false` by default.
```
Comment (aaron-contreras) [inline spec/support/edit_fields/edit_field.rb:24]: This is extremely valuable to have handy, thanks.

**Review by aaron-contreras — APPROVED** [review]
There's only one debugging leftover in a spec. Otherwise, looks good 👍

File: spec/features/work_packages/bulk/copy_work_package_spec.rb (hunk 2 above)
Comment (cbliard) [reply spec/features/work_packages/bulk/copy_work_package_spec.rb]: oups, yes, good catch :+1:

File: spec/support/edit_fields/edit_field.rb (hunk 4 above)
Comment (cbliard) [reply spec/support/edit_fields/edit_field.rb:24]: Yeah, most of it was generated by GitHub Copilot actually. I reviewed it and updated it where it made sense. It saves a lot of time about looking up what the exact RDoc syntax is.

All the doc added in `spec/support/pages/work_packages/work_packages_table.rb` has been generated this way.

**Pull request merged**